Swing Kit Docs (#1730)
vaadin/docs: articles/tools/swingkit/usage/exception.asciidoc

---
title: Exception Handling
order: 6

> layout: page

Reviewer: ```suggestion
description: How exceptions are handled when using Swing Kit, and how to configure it.
```

---

= Exception Handling

== Overview

> By default, any exception not handled by the remote View, will be sent to Swing to be handled by it.
> As a default behaviour, Swing will show the exception in a JOptionPane`.

Reviewer: ```suggestion
By default, any exception not handled by the remote View, is handled by Swing.
As a default behaviour, Swing shows the exception in a [classname]`JOptionPane`.
```

> The behaviour can be changed to catch any Vaadin exceptions from Swing, directly.

Reviewer: ```suggestion
This behaviour can be configured to catch any Vaadin exceptions from Swing, directly.
```

== Manually Handle Vaadin Exceptions

A custom handler for the exception can be used after disabling the default behaviour with the following:
[source,java]
----
vaadinPanel.disableDefaultExceptionHandler();
----

> You would add a custom handler for the exception by added this:

Reviewer: ```suggestion
Add a custom exception handler by adding the following:
```

[source,java]
----
vaadinPanel.addEventListener(EventType.EXCEPTION, event -> {
    // Print to standard output the exception content
    System.out.println(event.getParams().get("content"));
});
----

> Notice that the event will contain only `String` values instead of the original `Exception`.
> This is to avoid serialization errors due to different classpaths from the Vaadin server and the Swing application.

Reviewer: ```suggestion
Notice, that the event contains only [classname]`String` values instead of the original [classname]`Exception`.
This is to avoid serialization errors due to different classpaths from the Vaadin server and the Swing application.
```
